src/Decompressor/DecompressorGzip.php

```diff
@@ -25,9 +25,13 @@ public function decompress(SplFileInfo $sourceFile): void
                 ->setIdleTimeout(null)
                 ->mustRun();
         } catch (ProcessFailedException $e) {
```

Comment:
když je to stejně catch-all, tak bych to řízení těch výjimek nechal v komponentě a tím pádem by tady uvnitř nebyl logger ani ta gracefulness...

Reply:
Maximálně udělat výjimku DecompressionFailed a tu tam chytat.

Reply:
jj, to je vybornej napad. Presne tak jsem to mel a pak jsem si vsiml jedne drobnustky https://github.com/keboola/processor-decompress/blob/master/src/Component.php#L54 :)

Reply:
teda uplne presne ne, pojmenoval jsem tu vyjimku DecompressionException :)

Reply:
```
foreach ($finder as $sourceFile) {
try {
    $gzipDecompressor->decompress($sourceFile);
} catch (DecompressionException $e) {
    if (graceful) $logger->log(...) else throw $e
}
            }
```
?

Reply:
Nechápu :(

Reply:
decompress ma 4 volani a 3 implementace, takze kdyz to try-catch dam k implementaci tak je to jeden min vyskyt stejnyho bloku kodu nez kdyz ho dam k volani

ani tak ani tak to neni elegantni, ale zjevne to nejde napsat elegantne (bez nejakyho masivnejsiho refactoringu)

Reply:
Jo a proto mi přišlo lepší to napsat radši neelegantně ale odděleně, než o maličko míň neelegantně. IMHO ten decompressor nemá co vědět o tom, jestli se má chovat gracefully nebo ne. Ten prostě buď zvládne dekompresovat nebo ne. A když se to nepovede tak to má někam dát vědět a tam teprve se rozhodne co s tou informací udělat. Sám o sobě stejně nemůže ovlivnit, jestli se bude pokračovat nebo nebude, protože pracuje jen v kontextu jednoho souboru. 

Vypadá to, že ty obě řešení jsou ekvivalentní, ale IMHO nejsou. Především proto, že jedno z nich dělá větší coupling (přidává závislosti hlouběji). Jakmile vidím někde nějaké nové závislosti, tak je to vždycky red flag, jestli to nejde udělat bez toho...